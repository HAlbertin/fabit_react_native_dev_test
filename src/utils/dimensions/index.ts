import { Dimensions } from 'react-native';

const deviceWidth = Dimensions.get('window').width;
const deviceHeight = Dimensions.get('window').height;

const DimensionUtils = {
  deviceWidth,
  deviceHeight,
};

export default DimensionUtils;
