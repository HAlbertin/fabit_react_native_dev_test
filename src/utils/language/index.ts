import * as Localization from 'expo-localization';
import i18n from 'i18n-js';
import StorageUtils from '../storage';
import de from './translations/de';
import en from './translations/en';
import es from './translations/es';
import fr from './translations/fr';
import it from './translations/it';
import pt from './translations/pt';
import { TranslationKeys } from './translations/translations.interface';

export const getCurrentLocale = () => Localization.locale;

export const DEFAULT_LANGUAGE = 'en';
export const DEFAULT_TRANSLATION = {
  key: DEFAULT_LANGUAGE,
  value: 'English',
};

export const getCurrentLanguage = () => StorageUtils.getItem('USER_LANGUAGE') ?? DEFAULT_LANGUAGE;

export const changeLanguage = (newLocale: string) => {
  i18n.locale = newLocale;
};

export const translate = (key: TranslationKeys): string => i18n.t(key);

export const LANGUAGES = [
  { lang: 'pt', file: pt, name: 'Language_Pt' },
  { lang: 'en', file: en, name: 'Language_En' },
  { lang: 'es', file: es, name: 'Language_Es' },
  { lang: 'it', file: it, name: 'Language_It' },
  { lang: 'de', file: de, name: 'Language_De' },
  { lang: 'fr', file: fr, name: 'Language_Fr' },
];

const setupInitialLanguage = () => {
  i18n.fallbacks = true;

  // Set the key-value pairs for the different languages you want to support.
  i18n.translations = LANGUAGES.reduce((translations, { lang, file }) => {
    translations[lang] = file;
    return translations;
  }, {});

  // Set the locale once at the beginning of your app.
  i18n.locale = getCurrentLocale();
};

setupInitialLanguage();
