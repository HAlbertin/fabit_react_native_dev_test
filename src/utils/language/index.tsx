import * as Localization from 'expo-localization';
import i18n from 'i18n-js';
import de from './translations/de';
import en from './translations/en';
import es from './translations/es';
import fr from './translations/fr';
import it from './translations/it';
import pt from './translations/pt';
import { TranslationKeys } from './translations/translations.interface';

const LANGUAGES = [
  { lang: 'pt', file: pt },
  { lang: 'en', file: en },
  { lang: 'es', file: es },
  { lang: 'it', file: it },
  { lang: 'de', file: de },
  { lang: 'fr', file: fr },
];

// Set the key-value pairs for the different languages you want to support.
i18n.translations = LANGUAGES.reduce((translations, { lang, file }) => {
  translations[lang] = file;
  return translations;
}, {});

// Set the locale once at the beginning of your app.
i18n.locale = Localization.locale;

export const changeLanguage = (newLocale: string) => {
  i18n.locale = newLocale;
};

export const translate = (key: TranslationKeys): string => i18n.t(key);
