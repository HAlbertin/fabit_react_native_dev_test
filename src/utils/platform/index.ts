import { Platform } from 'react-native';

const isAndroid = () => Platform.OS === 'android';
const isIos = () => Platform.OS === 'ios';

const IOS = '3';
const ANDROID = '2';
const getPlatform = () => (PlatformUtils.isIos ? IOS : ANDROID);

const PlatformUtils = {
  isAndroid,
  isIos,
  getPlatform,
};

export default PlatformUtils;
