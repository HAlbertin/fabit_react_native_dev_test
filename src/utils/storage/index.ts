import { MMKV } from 'react-native-mmkv';
import StorageConstants from '../../constants/storage.constants';

const storage = new MMKV();

type storageKeys = keyof typeof StorageConstants;

const setItem = (keyName: storageKeys, value: string | number | boolean) => {
  storage.set(StorageConstants[keyName], value);
};

const getItem = (keyName: storageKeys) => storage.getString(StorageConstants[keyName]);

const deleteItem = (keyName: storageKeys) => storage.delete(keyName);

const StorageUtils = {
  setItem,
  getItem,
  deleteItem,
};

export default StorageUtils;
