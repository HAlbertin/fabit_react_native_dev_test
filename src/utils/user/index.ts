import StorageUtils from '../storage';

const isUserLoggedIn = () => !!StorageUtils.getItem('USER_SECRET');
const userNeedsVerify = () => !!StorageUtils.getItem('USER_SESSION_VERIFY');

const UserUtils = {
  isUserLoggedIn,
  userNeedsVerify,
};

export default UserUtils;
