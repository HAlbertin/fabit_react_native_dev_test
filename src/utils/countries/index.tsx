import countries from 'i18n-iso-countries';
import de from 'i18n-iso-countries/langs/de.json';
import en from 'i18n-iso-countries/langs/en.json';
import es from 'i18n-iso-countries/langs/es.json';
import fr from 'i18n-iso-countries/langs/fr.json';
import it from 'i18n-iso-countries/langs/it.json';
import pt from 'i18n-iso-countries/langs/pt.json';
import Flag from '../../components/flag';
import { getCurrentLanguage } from '../language';

const __setConfig = () => {
  countries.registerLocale(en);
  countries.registerLocale(es);
  countries.registerLocale(pt);
  countries.registerLocale(it);
  countries.registerLocale(fr);
  countries.registerLocale(de);
};

const GetAllCountries = () => {
  const countriesObj = countries.getNames(getCurrentLanguage(), { select: 'official' });
  const countriesList = Object.entries(countriesObj).map(e => ({
    key: e[0],
    value: e[1],
    icon: <Flag isoCode={e[0]} />,
  }));
  return countriesList;
};

__setConfig();

const CountriesUtils = {
  GetAllCountries,
};

export default CountriesUtils;
