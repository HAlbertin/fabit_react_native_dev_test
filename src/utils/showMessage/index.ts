import { Alert } from 'react-native';
import { translate } from '../language';
import { TranslationKeys } from '../language/translations/translations.interface';

const showMessage = (title: TranslationKeys, message?: TranslationKeys) =>
  Alert.alert(translate(title), message ? translate(message) : '');

const ShowMessageUtils = {
  showMessage,
};

export default ShowMessageUtils;
