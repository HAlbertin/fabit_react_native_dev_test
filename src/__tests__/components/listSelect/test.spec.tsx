import ListSelect from '../../../components/listSelect';
import { renderWithProviders } from '../../../utils/test';

describe('<ListSelect />', () => {
  const itemListMock = [
    {
      key: '1',
      value: 'teste',
    },
  ];

  it('should match snapshot', () => {
    const { toJSON } = renderWithProviders(<ListSelect listItems={itemListMock} onPress={jest.fn()} />);

    expect(toJSON()).toMatchSnapshot();
  });

  it('should render list items', () => {
    const { getByTestId } = renderWithProviders(<ListSelect listItems={itemListMock} onPress={jest.fn()} />);

    const renderedItem = getByTestId('list-item-1');

    expect(renderedItem).toBeTruthy();
  });
});
