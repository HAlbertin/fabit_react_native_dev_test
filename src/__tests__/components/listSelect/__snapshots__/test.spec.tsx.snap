// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListSelect /> should match snapshot 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "key": "1",
          "value": "teste",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    testID="list-select-flatlist"
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View>
          <View
            accessibilityState={
              Object {
                "disabled": false,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                false,
                Array [
                  Object {
                    "padding": 8,
                  },
                  undefined,
                ],
              ]
            }
            testID="list-item-1"
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "marginVertical": 6,
                      "paddingLeft": 8,
                    },
                    Object {
                      "flex": 1,
                      "justifyContent": "center",
                    },
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#0000",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "400",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Array [
                        Object {
                          "fontSize": 16,
                        },
                        Object {
                          "color": "rgba(0, 0, 0, 0.87)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  teste
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "rgba(0, 0, 0, 0.12)",
                  "height": 0.5,
                },
                undefined,
                undefined,
              ]
            }
          />
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
